community/fulltext-index/src/main/java/org/neo4j/kernel/api/impl/fulltext/LuceneFulltextDocumentStructure.java

```diff
@@ -141,12 +210,52 @@ private void setValues( Collection<String> names, Value[] values )
                 Value value = values[i++];
                 if ( value != null && value.valueGroup() == ValueGroup.TEXT )
                 {
-                    Field field = encodeValueField( name, value );
-                    document.add( field );
+                    addFulltextFieldsToDocument( name, value );
                 }
             }
         }
 
+        private void setValuesWithSort( Collection<String> propertyNames, Value[] values, Collection<String> sortProperties, Map<String,String> sortTypes )
+        {
+            int i = 0;
+
+            for ( String name : propertyNames )
+            {
+
+                Value value = values[i++];
+                if ( value != null && value.valueGroup() == ValueGroup.TEXT )
+                {
+                    addFulltextFieldsToDocument( name, value );
+                }
+            }
+
+            for ( String name : sortProperties )
+            {
+                Value value = values[i++];
+                if ( value != null )
+                {
+                    Field sortableField = encodeSortableValueField( name, value );
+                    if ( sortableField != null && validateSortType( sortableField, sortTypes.get( name ) ) )
```

Comment:
If the type is not one of `STRING`, `DOUBLE`, or `LONG`, this will just silently not add the field.
Is this the desired behavior, or should it at least log out a warning?

Reply:
If we do log out a warning to the debug log, it could get very chatty. I don't like the idea of just ignoring by default though. We can't control mixed-schema issues but we should find a way to notify.

@alexiudice Let's add this to the design document for further considerations and leave it as-is for now.

Reply:
Ok sounds good. The current behavior is to silently not add the field. As far as I know there is no access to logging out to the debug log from that class.

Reply:
@bradnussbaum @ThomasMatlak 

I just found that a logging object is provided in the class `FulltextIndexProvider`, which seems to serve as a unifying class that most of the other Fulltext classes are in some way connected to. We could theoretically pass in the log from `FulltextProvider` to the other classes and then be able to log out messages if we wanted to add that.